flink-connectors/flink-connector-pulsar/src/test/java/org/apache/flink/connector/pulsar/table/catalog/PulsarCatalogITCase.java

```diff
@@ -770,6 +773,64 @@ void copyDataFromNativeTableToNativeTable() throws Exception {
             assertThat(result).containsExactlyElementsOf(INTEGER_LIST);
         }
 
+        @Test
+        void readFromNativeTableWithKeyValueAvroSchema() throws Exception {
+            String topicName = newTopicName();
+            Schema<TestingUser> schema = Schema.AVRO(TestingUser.class);
+            KeyValue<TestingUser, TestingUser> expectedUser =
+                    new KeyValue<>(createRandomUser(), createRandomUser());
+            Schema<KeyValue<TestingUser, TestingUser>> keyValueSchema =
+                    KeyValueSchemaImpl.of(schema, schema, KeyValueEncodingType.SEPARATED);
+
+            pulsar.operator().sendMessage(topicName, keyValueSchema, expectedUser);
+
+            tableEnv.useCatalog(PULSAR_CATALOG1);
+            tableEnv.useDatabase(PULSAR1_DB);
+
+            final List<Row> result =
+                    collectRows(
+                            tableEnv.sqlQuery(
+                                    String.format(
+                                            "select age, name, `key.age`, `key.name` from %s",
+                                            TopicName.get(topicName).getLocalName())),
+                            1);
+            assertThat(result)
+                    .containsExactlyElementsOf(
+                            Collections.singletonList(
+                                    Row.of(
+                                            expectedUser.getValue().getAge(),
+                                            expectedUser.getValue().getName(),
+                                            expectedUser.getKey().getAge(),
+                                            expectedUser.getKey().getName())));
+        }
+
+        @Test
+        void readFromNativeTableWithKeyValuePrimitiveSchema() throws Exception {
```

Comment:
can you also add the test case for KeyValue<String, AVRO> ?

Reply:
done